source/Engine.cpp

```diff
@@ -100,10 +100,36 @@ namespace {
 		return Radar::UNFRIENDLY;
 	}
 
-	constexpr auto PrunePointers = [](auto &objects) { erase_if(objects,
-			[](const auto &obj) { return obj->ShouldBeRemoved(); }); };
-	constexpr auto Prune = [](auto &objects) { erase_if(objects,
-			[](const auto &obj) { return obj.ShouldBeRemoved(); }); };
```

Comment:
i think it might make sense to keep these in a comment for the day when we actually do the switch(when enough support), cause I really like them.